File: aspnet-core/src/HC.WeChat.Core/WinningRecords/WinningRecord.cs
using Abp.Domain.Entities;
using Abp.Domain.Entities.Auditing;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace HC.WeChat.WinningRecords
{
    /// <summary>
    /// 中奖纪录
    /// </summary>
    [Table("WinningRecords")]
    public class WinningRecord : Entity<Guid>, IHasCreationTime
    {

        /// <summary>
        /// 奖品Id 外键
        /// </summary>
        public virtual Guid? PrizeId { get; set; }

        /// <summary>
        /// 用户Id 外键
        /// </summary>
        public virtual Guid? UserId { get; set; }

        /// <summary>
        /// 收货地址Id 外键
        /// </summary>
        public virtual Guid? AddressId { get; set; }

        /// <summary>
        /// 中奖时间
        /// </summary>
        public virtual DateTime? WinningTime { get; set; }

        /// <summary>
        /// 中奖数量
        /// </summary>
        public virtual int? Num { get; set; }

        /// <summary>
        /// 过期时间
        /// </summary>
        [Required]
        public virtual DateTime ExpiryTime { get; set; }

        /// <summary>
        /// 状态（未兑换、已申领、已兑换、已过期 枚举）
        /// </summary>
        public virtual int? Status { get; set; }

        /// <summary>
        /// 申领时间
        /// </summary>
        public virtual DateTime? ApplyTime { get; set; }

        /// <summary>
        /// 兑换时间
        /// </summary>
        public virtual DateTime? CompleteTime { get; set; }

        /// <summary>
        /// 快递公司
        /// </summary>
        [StringLength(200)]
        public virtual string ExpressCompany { get; set; }

        /// <summary>
        /// 快递单号
        /// </summary>
        [StringLength(50)]
        public virtual string ExpressNo { get; set; }

        /// <summary>
        /// CreationTime
        /// </summary>
        [Required]
        public virtual DateTime CreationTime { get; set; }

        /// <summary>
        /// CreatorUserId
        /// </summary>
        public virtual long? CreatorUserId { get; set; }
    }
}

File: ng-admin/src/app/shared/entity/member/winningrecord.ts
export class WinningRecord implements IWinningRecord {
    id: string;
    prizeId: string;
    userId: string;
    addressId: string;
    winningTime: Date;
    num: number;
    expiryTime: Date;
    status: number;
    applyTime: Date;
    completeTime: Date;
    expressCompany: string;
    expressNo: string;
    creationTime: Date;
    creatorUserId: number;
    prizeName: string;
    userName: string;
    statusName: string;
    constructor(data?: IWinningRecord) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }

    init(data?: any) {
        if (data) {
            this.id = data["id"];
            this.prizeId = data["prizeId"];
            this.userId = data["userId"];
            this.addressId = data["addressId"];
            this.winningTime = data["winningTime"];
            this.num = data["num"];
            this.expiryTime = data["expiryTime"];
            this.status = data["status"];
            this.applyTime = data["applyTime"];
            this.completeTime = data["completeTime"];
            this.expressCompany = data["expressCompany"];
            this.expressNo = data["expressNo"];
            this.creationTime = data["creationTime"];
            this.creatorUserId = data["creatorUserId"];
            this.prizeName = data["prizeName"];
            this.userName = data["userName"];
            this.statusName = data["statusName"];
        }
    }

    static fromJS(data: any): WinningRecord {
        let result = new WinningRecord();
        result.init(data);
        return result;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["id"] = this.id;
        data["prizeId"] = this.prizeId;
        data["userId"] = this.userId;
        data["addressId"] = this.addressId;
        data["winningTime"] = this.winningTime;
        data["num"] = this.num;
        data["expiryTime"] = this.expiryTime;
        data["status"] = this.status;
        data["applyTime"] = this.applyTime;
        data["completeTime"] = this.completeTime;
        data["expressCompany"] = this.expressCompany;
        data["expressNo"] = this.expressNo;
        data["creationTime"] = this.creationTime;
        data["creatorUserId"] = this.creatorUserId;
        data["prizeName"] = this.prizeName;
        data["userName"] = this.userName;
        data["statusName"] = this.statusName;
        return data;
    }

    clone() {
        const json = this.toJSON();
        let result = new WinningRecord();
        result.init(json);
        return result;
    }
}

export interface IWinningRecord {
    id: string;
    prizeId: string;
    userId: string;
    addressId: string;
    winningTime: Date;
    num: number;
    expiryTime: Date;
    status: number;
    applyTime: Date;
    completeTime: Date;
    expressCompany: string;
    expressNo: string;
    creationTime: Date;
    creatorUserId: number;
    prizeName: string;
    userName: string;
    statusName: string;
}